MvvmCross/Droid/Droid/Views/MvxAndroidViewPresenter.cs

```diff
@@ -236,39 +240,42 @@ protected Type GetCurrentActivityViewModelType()
             if (attribute.ActivityHostViewModelType == null)
                 attribute.ActivityHostViewModelType = GetCurrentActivityViewModelType();
 
-            Task.Run(async () => {
-                await ShowHostActivity(attribute);
-            }).ContinueWith((result) => {
-                Mvx.Resolve<IMvxMainThreadDispatcher>().RequestMainThreadAction(() => {
-                    if (CurrentActivity.FindViewById(attribute.FragmentContentId) == null)
-                        throw new NullReferenceException("FrameLayout to show Fragment not found");
+            var currentHostViewModelType = GetCurrentActivityViewModelType();
+            if (attribute.ActivityHostViewModelType != currentHostViewModelType)
+            {
+                _pendingRequest = request;
```

Comment:
Can we make it more stateless ? I mean to put data to activity intent and then get intent data and invoke event with request. I'm worry about this filed.